src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,107 @@
+/* eslint-isInvalid @typescript-eslint/no-unused-vars */
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+// import { event } from 'cypress/types/jquery';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  function resetForm() {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  }
+
+  const isInvalid = !title || !imgUrl || !imdbUrl || !imdbId;
+
+  const handleSubmit = (value: React.FormEvent) => {
+    value.preventDefault();
+
+    if (isInvalid) {
+      return;
+    }
+
+    onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    });
+
+    setCount(counter => counter + 1);
+    resetForm();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(value) => {
+          setTitle(value);
+        }}
```

Comment:
```suggestion
        onChange={setTitle}
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**